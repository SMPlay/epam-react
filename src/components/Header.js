import React from "react";
import "./Header.css";

const Header = () => {
  return (
    <header>
      <nav>
        <ul>
          <li>Li 1</li>
          <li>Li 2</li>
          <li>Li 3</li>
          <li>Li 4</li>
          <li>Li 5</li>
        </ul>
      </nav>
    </header>
  )
}

export default Header;