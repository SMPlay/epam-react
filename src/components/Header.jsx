import React from "react";
import PropTypes from "prop-types";
import { createUseStyles } from "react-jss";
import bg from "../img/header_background.jpg";

const useStyles = createUseStyles({
  headerBg: {
    position: "relative",
    backgroundImage: `url(${bg})`,
    backgroundSize: "cover",
    backgroundPosition: "center",
    "&::before": {
      content: "''",
      position: "absolute",
      width: "100%",
      height: "100%",
      background: "black",
      opacity: ".4",
      zIndex: 0,
    },
  },
});

const Header = ({ component }) => {
  const classes = useStyles();

  return (
    <header className={classes.headerBg}>
      <div className="container">
        {component}
      </div>
    </header>
  );
};

Header.propTypes = {
  component: PropTypes.node.isRequired,
};

export default Header;
