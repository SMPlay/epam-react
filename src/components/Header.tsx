import React, { ReactNode } from "react";
import PropTypes from "prop-types";
import { createUseStyles } from "react-jss";
import bg from "../img/header_background.jpg";

const useStyles = createUseStyles({
  headerBg: {
    position: "relative",
    backgroundImage: `url(${bg})`,
    backgroundSize: "cover",
    backgroundPosition: "center",
    "&::before": {
      content: "''",
      position: "absolute",
      width: "100%",
      height: "100%",
      background: "black",
      opacity: ".4",
      zIndex: 0,
    },
  },
});

interface Props {
  children: ReactNode
}

export const Header: React.FC<Props> = ({ children }) => {
  const classes = useStyles();

  return (
    <header className={classes.headerBg}>
      <div className="container">
        {children}
      </div>
    </header>
  );
};

Header.propTypes = {
  children: PropTypes.node.isRequired,
};
