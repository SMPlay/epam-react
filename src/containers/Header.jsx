import React from "react";
import { createUseStyles } from "react-jss";
import Wrapper from "components/Wrapper";
import Logo from "components/Logo";
import Search from "./Search";
import bg from "../img/header_background.jpg";

const useStyles = createUseStyles({
  headerBg: {
    position: "relative",
    backgroundImage: `url(${bg})`,
    backgroundSize: "cover",
    backgroundPosition: "center",
    "&::before": {
      content: "''",
      position: "absolute",
      width: "100%",
      height: "100%",
      background: "black",
      opacity: ".4",
      zIndex: 0,
    },
  },
});

const Header = () => {
  const classes = useStyles();

  return (
    <header className={classes.headerBg}>
      <Wrapper padding={20}>
        <Logo />
        <Search />
      </Wrapper>
    </header>
  );
};

export default Header;
