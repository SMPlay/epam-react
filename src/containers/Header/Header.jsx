import React from "react";
import { createUseStyles } from "react-jss";
import Container from "components/Container";
import Search from "./Search/Search";
import bg from "../../img/header_background.jpg";

const useStyles = createUseStyles({
  headerBg: {
    position: "relative",
    backgroundImage: `url(${bg})`,
    backgroundSize: "cover",
    backgroundPosition: "center",
    "&::before": {
      content: "''",
      position: "absolute",
      width: "100%",
      height: "100%",
      background: "black",
      opacity: ".4",
      zIndex: 0,
    },
  },
  title: {
    fontSize: 18,
    color: "rgb(250, 19, 19)",
  },
});

const Header = () => {
  const classes = useStyles();

  return (
    <header className={classes.headerBg}>
      <Container padding={20}>
        <h1 className={classes.title}>netflixroulette</h1>
        <Search />
      </Container>
    </header>
  );
};

export default Header;
