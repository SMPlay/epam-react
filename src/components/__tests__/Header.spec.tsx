import React from "react";
import renderer from "react-test-renderer";
import { BrowserRouter as Router } from "react-router-dom";
import { Header } from "../Header";

describe("Header component", () => {
  it("should render with props", () => {
    const tree = renderer.create(
      <Router>
        <Header>
          <div />
        </Header>
      </Router>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
