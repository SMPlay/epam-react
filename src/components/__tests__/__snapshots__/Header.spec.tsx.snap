// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should render with props 1`] = `
<header
  className="headerBg-0-2-1"
>
  <div
    className="container"
  >
    <div />
  </div>
</header>
`;
